proto/osmosis/cosmwasmpool/v1beta1/model/module_sudo_msg.proto

```diff
@@ -0,0 +1,81 @@
+syntax = "proto3";
+package osmosis.cosmwasmpool.v1beta1;
+
+import "gogoproto/gogo.proto";
+import "cosmos/base/v1beta1/coin.proto";
+
+option go_package = "github.com/osmosis-labs/osmosis/v15/x/cosmwasmpool/cosmwasm/msg";
+
+// ===================== SwapExactAmountIn
+message SwapExactAmountIn {
+  string sender = 1;
+  // token_in is the token to be sent to the pool.
+  cosmos.base.v1beta1.Coin token_in = 2 [ (gogoproto.nullable) = false ];
+
+  // token_out_denom is the token denom to be received from the pool.
+  string token_out_denom = 3;
+
+  // token_out_min_amount is the minimum amount of token_out to be received from
+  // the pool.
+  string token_out_min_amount = 4 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable) = false
+  ];
+
+  // swap_fee is the swap fee for this swap estimate.
+  string swap_fee = 5 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+}
+
+message SwapExactAmountInSudoMsg {
+  // swap_exact_amount_in is the structure containing all the request
+  // information for this message.
+  SwapExactAmountIn swap_exact_amount_in = 1 [ (gogoproto.nullable) = false ];
+}
+
+message SwapExactAmountInSudoMsgResponse {
+  // token_out_amount is the token out computed from this swap estimate call.
+  string token_out_amount = 1 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable) = false
+  ];
+}
+
+// ===================== SwapExactAmountOut
+message SwapExactAmountOut {
+  string sender = 1;
+  // token_out is the token to be sent out of the pool.
+  cosmos.base.v1beta1.Coin token_out = 2 [ (gogoproto.nullable) = false ];
+
+  // token_in_denom is the token denom to be sent too the pool.
+  string token_in_denom = 3;
+
+  // token_in_max_amount is the maximum amount of token_in to be sent to the
+  // pool.
+  string token_in_max_amount = 4 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable) = false
+  ];
+
+  // swap_fee is the swap fee for this swap estimate.
+  string swap_fee = 5 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+}
+
+message SwapExactAmountOutSudoMsg {
```

Comment:
If we're gonna launch the contract first and pools chain side later, then the messages need to be execute, not sudo (as only the chain can send sudo messages).

Reply:
Ideally, this should still be a sudo message, and the contract refactored and upgraded.

We should be customizing the contract to make the chain code as secure and general as possible instead of adjusting chain code to be transmuter-specific

Reply:
https://github.com/osmosis-labs/osmosis/issues/4797

Reply:
I still think it makes sense for this to be an exec. Is there a reason we don't want people using the pools directly?

Reply:
Tagged you here with the reasoning: https://github.com/osmosis-labs/osmosis/pull/4810#discussion_r1154605426

Even if we decide to change this in the future, let's not block this PR on this request, please. The reason is that everything else in the spike #4675, including the test contract, depends on these messages being sudo. As a result, it would be blocked on Boss refactoring the test contract.

I will make sure to use #4797 as a tracker to address this based on the conclusions from discussions we are having here: https://github.com/osmosis-labs/osmosis/pull/4810#discussion_r1154605426

Let me know if that works